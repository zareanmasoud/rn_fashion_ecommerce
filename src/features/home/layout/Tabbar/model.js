import R from 'res/R';

const model = {
  buttons: {
    home: {
      id: 0,
      handlePress: () => {},
      src: R.images.backButton,
      text: R.strings.home.layout.mainTabbar.home,
    },
    wishlist: {
      id: 1,
      handlePress: () => {},
      src: R.images.cartButton,
      text: R.strings.home.layout.mainTabbar.wishlist,
    },
    help: {
      id: 2,
      handlePress: () => {},
      src: R.images.logoButton,
      text: R.strings.home.layout.mainTabbar.help,
    },
    profile: {
      id: 3,
      handlePress: () => {},
      src: R.images.logoButton,
      text: R.strings.home.layout.mainTabbar.profile,
    },
  },
};

export default model;
