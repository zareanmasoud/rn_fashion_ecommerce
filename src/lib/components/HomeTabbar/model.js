// @flow

export type Buttons = {
  press: Function,
  src: string,
  text: string,
};
