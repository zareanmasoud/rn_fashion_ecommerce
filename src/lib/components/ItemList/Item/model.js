// @flow

export type ItemModel = {
  id: string,
  photo: string,
};
