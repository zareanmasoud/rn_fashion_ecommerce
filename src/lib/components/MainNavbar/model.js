// @flow

export type Button = {
  action: Function,
  src: string,
};
