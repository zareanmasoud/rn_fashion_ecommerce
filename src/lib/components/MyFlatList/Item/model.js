// @flow

export type Data = {
  id: string,
  photo: string,
};
