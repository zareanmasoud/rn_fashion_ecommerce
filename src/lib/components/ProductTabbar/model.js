// @flow

export type Specs = {
  action: Function,
  src: string,
  text: string,
};
