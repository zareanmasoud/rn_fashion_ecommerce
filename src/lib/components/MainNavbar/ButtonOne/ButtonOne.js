import React from 'react';
import {View} from 'react-native';
import NavbarButton from '../NavbarButton';
import styles from '../styles';
import type {Button} from '../types';

type Props = {
  backButton: Button,
  logoButton: Button,
};

export default function ButtonOne({backButton, logoButton}: Props) {
  if (backButton === undefined) {
    if (logoButton) return <NavbarButton {...logoButton} logoMode />;
    return <View style={styles.emptyView} />;
  }
  return <NavbarButton {...backButton} />;
}
