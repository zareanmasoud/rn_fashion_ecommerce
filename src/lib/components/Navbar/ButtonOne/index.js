import ButtonOne from './ButtonOne';

export {ButtonOne as default};
