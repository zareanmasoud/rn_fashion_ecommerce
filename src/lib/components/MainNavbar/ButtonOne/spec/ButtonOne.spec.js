import React from 'react';
import {shallow} from 'enzyme';
import ButtonOne from '../ButtonOne';
import model from './model';

describe('ButtonOne Component', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const wrapper = shallow(<ButtonOne {...model.buttons} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
