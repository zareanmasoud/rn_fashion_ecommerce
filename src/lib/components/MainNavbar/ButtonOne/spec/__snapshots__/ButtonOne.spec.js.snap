// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonOne Component Rendering should match to snapshot 1`] = `
<NavbarButton
  handlePress={[Function]}
  src={
    Object {
      "process": [Function],
    }
  }
/>
`;
